doc/guides/style_guide.md

```diff
@@ -0,0 +1,473 @@
+# Style Guidelines
```

Comment:
Consider changing to `Python Style Guidelines` and renaming the file to `style_guide_python.md`